src/BaseSchema.hack

```diff
@@ -21,4 +25,24 @@ abstract class BaseSchema {
     ): Awaitable<ValidFieldResult<?dict<string, mixed>>> {
         return new ValidFieldResult(null);
     }
+
+    <<__Override>>
+    final public function getQueryType(): Introspection\__Type {
+        $query_type = static::QUERY_TYPE;
+        return $query_type::nonNullable();
+    }
+
+    <<__Override>>
+    final public function getMutationType(): ?Introspection\__Type {
+        $mutation_type = static::MUTATION_TYPE;
+        return $mutation_type is nonnull ? $mutation_type::nonNullable() : null;
+    }
+
+    final public function getType(string $name): ?Introspection\__Type {
+        $types = Dict\merge(static::INPUT_TYPES, static::OUTPUT_TYPES);
```

Comment:
Should pull this out and memoize it

Reply:
Why not just

```Hack
$type = static::INPUT_TYPES[$name] ?? static::OUTPUT_TYPES[$name] ?? null;
```

instead of constructing the big map?

Reply:
Ah good point, will fix this